job-runner/runner/runner.go

```diff
@@ -35,12 +35,15 @@ type Runner interface {
 
 // A runner represents all information needed to run a job.
 type runner struct {
-	pJob    proto.Job // struct with info about the job
 	realJob job.Job   // the actual job interface to run
 	reqId   string    // the request id the job belongs to
 	rmc     rm.Client // client used to send JLs to the RM
 	// --
-	stopChan chan struct{} // used on Stop
+	jobId     string
+	jobType   string
+	maxTries  uint
+	retryWait time.Duration
```

Comment:
It seems like `jobId`, `jobType`, `maxTries`, and `retryWait` are all in `proto.Job`. Are we listing them out here rather than use the `proto.Job` because we don't need to reference all of the fields in the struct or because we want to capture the data at this point in time?

Reply:
Replied elsewhere, but in short: it decouples proto.Job from runner internals, allowing us to change proto.Job and then have only one place to update runner, rather than having to update runner's direct use of proto.Job throughout its internals.